src/services/services.ts

```diff
@@ -534,25 +542,30 @@ function createSyntaxList(nodes: NodeArray<Node>, parent: Node): Node {
     return list;
 }
 
-class TokenOrIdentifierObject implements Node {
-    public kind!: SyntaxKind;
+class TokenOrIdentifierObject<TKind extends SyntaxKind> implements Node {
+    public kind: TKind;
     public pos: number;
     public end: number;
     public flags: NodeFlags;
-    public modifierFlagsCache: ModifierFlags;
+    public modifierFlagsCache!: ModifierFlags;
     public transformFlags: TransformFlags;
     public parent: Node;
     public symbol!: Symbol;
     public jsDocComments?: JSDoc[];
+    public id?: number;
+    public emitNode?: EmitNode | undefined;
 
-    constructor(pos: number, end: number) {
+    constructor(kind: TKind, pos: number, end: number) {
         // Set properties in same order as NodeObject
+        // Note: must match Token and Identifier in src/compiler/utilities.ts
         this.pos = pos;
         this.end = end;
+        this.kind = kind;
+        this.id = 0;
         this.flags = NodeFlags.None;
-        this.modifierFlagsCache = ModifierFlags.None;

```

Comment:
The compiler does not set this prop at all. I suspect I could just leave this and get the same effect without the baseline change, at a cost of memory.

Reply:
We don't set it because tokens and identifiers can't have modifiers, so we shave off some memory by ignoring it.

Reply:
Yeah, that's what I assumed, though I was surprised given it means that the shape of `Node` is different. But I guess they just don't go down the same paths? Probably worth another investigation (unless you've already done it) as this PR is just trying to copy the compiler for now.